File: src/main/java/net/softbell/bsh/dto/view/advance/NodeInfoCardDto.java
package net.softbell.bsh.dto.view.advance;

import java.util.Date;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.EnableStatusRule;
import net.softbell.bsh.domain.entity.Node;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 노드 수정뷰 정보 카드정보 DTO
 */
@Getter
@Setter
public class NodeInfoCardDto
{
	private Long nodeId;
	private EnableStatusRule enableStatus;
	private String nodeName;
	private String alias;
	private String uid;
	private String token;
	private Byte controlMode;
	private String version;
	private Date registerDate;
	private Date approvalDate;
	
	public NodeInfoCardDto(Node entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Convert
		this.nodeId = entity.getNodeId();
		this.enableStatus = entity.getEnableStatus();
		this.nodeName = entity.getNodeName();
		this.alias = entity.getAlias();
		this.uid = entity.getUid();
		this.token = entity.getToken();
		this.controlMode = entity.getControlMode();
		this.version = entity.getVersion();
		this.registerDate = entity.getRegisterDate();
		this.approvalDate = entity.getApprovalDate();
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/advance/NodeInfoCardDto.kt
package net.softbell.bsh.dto.view.advance

import net.softbell.bsh.domain.EnableStatusRule
import net.softbell.bsh.domain.entity.Node
import java.util.*

/**
 * @author : Bell(bell@softbell.net)
 * @description : 노드 수정뷰 정보 카드정보 DTO
 */
class NodeInfoCardDto(entity: Node) {
    val nodeId: Long = entity.nodeId
    val enableStatus: EnableStatusRule = entity.enableStatus
    val nodeName: String = entity.nodeName
    val alias: String = entity.alias
    val uid: String = entity.uid
    val token: String? = entity.token
    val controlMode: Byte = entity.controlMode
    val version: String = entity.version
    val registerDate: Date = entity.registerDate
    val approvalDate: Date? = entity.approvalDate
}